src/custom/conversions.rs

```diff
@@ -0,0 +1,463 @@
+    #[case("trunc_s", 5)]
+    fn test_truncation(mut llvm_ctx: TestContext, #[case] op_name: &str, #[case] width: u8) -> () {
+        llvm_ctx.add_extensions(add_int_extensions);
+        llvm_ctx.add_extensions(add_float_extensions);
+        llvm_ctx.add_extensions(add_conversions_extension);
+        llvm_ctx.add_extensions(add_default_prelude_extensions);
+        let in_ty = FLOAT64_TYPE;
+        let out_ty = sum_with_error(INT_TYPES[width as usize].clone());
+        let hugr = test_conversion_op(op_name, in_ty, out_ty.into(), width);
+        check_emission!(op_name, hugr, llvm_ctx);
+    }
+
+    #[rstest]
+    fn my_test_exec(mut exec_ctx: TestContext) {
+        let hugr = SimpleHugrConfig::new()
+            .with_outs(USIZE_T)
+            .with_extensions(PRELUDE_REGISTRY.to_owned())
+            .finish(|mut builder: DFGW| {
+                let konst = builder.add_load_value(ConstUsize::new(42));
+                builder.finish_with_outputs([konst]).unwrap()
+            });
+        exec_ctx.add_extensions(add_default_prelude_extensions);
+        assert_eq!(42, exec_ctx.exec_hugr_u64(hugr, "main"));
+    }
+
+    #[rstest]
+    #[case(0)]
+    #[case(42)]
+    #[case(18_446_744_073_709_551_615)]
+    fn usize_roundtrip(mut exec_ctx: TestContext, #[case] val: u64) -> () {
+        let hugr = SimpleHugrConfig::new()
+            .with_outs(USIZE_T)
+            .with_extensions(CONVERT_OPS_REGISTRY.clone())
+            .finish(|mut builder: DFGW| {
+                let k = builder.add_load_value(ConstUsize::new(val));
+                let [int] = builder
+                    .add_dataflow_op(ConvertOpDef::ifromusize.without_log_width(), [k])
+                    .unwrap()
+                    .outputs_arr();
+                let [usize_] = builder
+                    .add_dataflow_op(ConvertOpDef::itousize.without_log_width(), [int])
+                    .unwrap()
+                    .outputs_arr();
+                builder.finish_with_outputs([usize_]).unwrap()
+            });
+        exec_ctx.add_extensions(add_int_extensions);
+        exec_ctx.add_extensions(add_conversions_extension);
+        exec_ctx.add_extensions(add_default_prelude_extensions);
+        assert_eq!(val, exec_ctx.exec_hugr_u64(hugr, "main"));
+    }
+
+    #[rstest]
+    // Being roundtrippable from float isn't defined on every value in u64, but
+    // here are some that should work.
+    #[case(0)]
+    #[case(3)]
+    #[case(255)]
+    #[case(4294967295)]
+    #[case(42)]
+    #[case(18_000_000_000_000_000_000)]
+    // N.B.: There's some strange behaviour at the upper end of the ints - the
+    // first case gets converted to something that's off by 1,000, but the second
+    // (which is (2 ^ 64) - 1) gets converted to (2 ^ 32) - off by 9 million!
+    // The fact that the first case works as expected  tells me this isn't to do
+    // with int widths - maybe a floating point expert could explain that this
+    // is standard behaviour...
```

Comment:
Any int less than 2^53 can be exactly expressed as a `f64` since there are 52 mantissa bits.  Besides that you can represent these numbers scaled by powers of two. So I'd expect `2^64 - 1` to be truncated to `2^63 = 2^53 * 2^9`??